conans/test/functional/toolchains/cmake/test_cmake_toolchain.py

```diff
@@ -436,3 +438,30 @@ def test_autolink_pragma_without_components(self):
                      "test_package/conanfile.py": self.test_cf})
 
         client.run("create .")
+
+
+@pytest.mark.skipif(platform.system() != "Windows", reason="Only for windows")
+def test_cmake_toolchain_runtime_types():
+    # everything works with the default cmake_minimum_required version 3.15 in the template
+    client = TestClient(path_with_spaces=False)
+    client.run("new hello/0.1 --template=cmake_lib")
+    client.run("install . -s compiler.runtime=MTd -s build_type=Debug")
+    client.run("build .")
+
+    vcvars = vcvars_command(version="15", architecture="x64")
+    lib = os.path.join(client.current_folder, "build", "Debug", "hello.lib")
+    dumpbind_cmd = '{} && dumpbin /directives "{}"'.format(vcvars, lib)
+    client.run_command(dumpbind_cmd)
+    assert "LIBCMTD" in client.out
+
+    client.save({}, clean_first=True)
+    # Setting an older cmake_minimum_required in the CMakeLists fails, will link
+    # against the default debug runtime (MDd->MSVCRTD), not against MTd->LIBCMTD
+    client.run("new hello/0.1 --template=cmake_lib")
+    replace_in_file(os.path.join(client.current_folder, "CMakeLists.txt"),
+                    'cmake_minimum_required(VERSION 3.15)',
+                    'cmake_minimum_required(VERSION 3.1)', output=client.out)
```

Comment:
Isn't it possible that 3.1 is too old, and something else is broken? I think I tried with 3.10 or something like that.

Reply:
It also fails for me with 3.14, I'll update the test to use a newer version

Reply:
I made it work by dropping CMAKE_PROJECT_INCLUDE, and defining the policy directly in the CMakePresets file.

Apparently, when interacting with cmake_minimum_required(), the policy would need to be set before ``project()`` and that means ``CMAKE_PROJECT_INCLUDE_BEFORE``. But it happens that the toolchain is not "activated" until after the ``project()``, so all attempts to define it in the toolchain are doomed. It is not possible.

Reply:
Maybe we should check the policy in the toolchain and fail if not set to `NEW` so that we make sure that if someone calls the toolchain directly the runtime type is properly set?

Reply:
I will try to add it, lets see if possible.

Reply:
https://github.com/conan-io/conan/pull/11098/commits/3719c2c5fa521b67f817bec0efb91bd149b1abdd seems to do the job